src/Operations.jl

```diff
@@ -852,65 +853,78 @@ function download_artifacts(ctx::Context;
+    end
+    for (artifacts_toml, artifacts) in all_collected_artifacts
+        # For each Artifacts.toml, install each artifact we've collected from it
+        for name in keys(artifacts)
+            local rname = rpad(name, longest_name_length)
+            local hash = SHA1(artifacts[name]["git-tree-sha1"])
+            local bar = MiniProgressBar(;header=rname, main=false, indent=2, color = Base.info_color(), mode=:data, always_reprint=true)
+            local dstate = DownloadState(:ready, "", time_ns(), Base.ReentrantLock(), bar)
+            function progress(total, current; status="")
+                local t = time_ns()
+                if isempty(status)
+                    dstate.bar.max = total
+                    dstate.bar.current = current
+                end
+                lock(dstate.status_lock) do
+                    dstate.status = status
+                    dstate.status_update_time = t
+                end
+            end
+            # returns a string if exists, or function that downloads the artifact if not
+            local ret = ensure_artifact_installed(name, artifacts[name], artifacts_toml;
+                                        verbose, quiet_download=!(usable_io(io)), io, progress)
+            if ret isa Function
+                download_states[hash] = dstate
+                download_jobs[hash] =
+                    () -> begin
+                        try
+                            dstate.state = :running
+                            ret()
+                            if !fancyprint
+                                @lock print_lock printpkgstyle(io, :Installed, "artifact $rname $(MiniProgressBars.pkg_format_bytes(dstate.bar.max; sigdigits=1))")
                             end
+                        catch
+                            dstate.state = :failed
+                            rethrow()
+                        else
+                            dstate.state = :done
                         end
-                    )
-                end
+                    end
             end
-            push!(used_artifact_tomls, artifacts_toml)
         end
     end
 
     if !isempty(download_jobs)
-        longest_name_length = maximum(textwidth, keys(download_states))
-        for (name, dstate) in download_states
-            dstate.bar.header = rpad(name, longest_name_length)
-        end
-
         if fancyprint
             t_print = Threads.@spawn begin
                 try
                     print(io, ansi_disablecursor)
                     first = true
                     timer = Timer(0, interval=1/10)
                     # TODO: Implement as a new MiniMultiProgressBar
-                    main_bar = MiniProgressBar(; indent=1, header = "Downloading artifacts", color = :green, mode = :int, always_reprint=true)
+                    main_bar = MiniProgressBar(; indent=1, header = "Installing artifacts", color = :green, mode = :int, always_reprint=true)
```

Comment:
```suggestion
                    main_bar = MiniProgressBar(; indent=2, header = "Installing artifacts", color = :green, mode = :int, always_reprint=true)
```
Given `Installing` is one shorter. To align right